### ISSUE TITLE
no-duplicate-param fails on function with two es6-destructured params and no named params

### ISSUE THREAD

**Issue by hawkrives** [eslint/eslint#2148]
This code:

```js
function organizeValues([key, values], {words=false, profWords=false}={}) {
	let organizedValues = values.map(val => {
		if (val.startsWith('$')) {
			return val.toUpperCase()
		}
		// there's more stuff here
	})

	return [key, organizedValues]
}
```

produces this warning (among others):

```
<text>
  1:0   error    Duplicate param 'undefined'                    no-dupe-args
```

It shouldn't be producing any duplicate parameter warnings.

---

`eslint -v`: v0.17.1

**Comment by nzakas** [comment]
Oops, looks like the rule is confused by destructed params. We will get this fixed.

**Comment by hawkrives** [comment]
Thanks!

**Comment by nzakas** [comment]
Working on this.

**Comment by rdy** [comment]
This still fails if there are two levels of destructuring in the functions arguments.

```
function foo([{a}, {b}]) {
  return a + b;
}
```
Eslint fails with:
```
Message:
    Duplicate param 'undefined'.
```

eslint -v: v0.19.0

**Comment by nzakas** [comment]
Please file a new issue
